components/codec/src/number.rs

```diff
@@ -917,6 +917,15 @@ pub trait NumberEncoder: BufferWriter {
         write!(self, v, 2, encode_u16_le)
     }
 
+    /// Writes an unsigned 16 bit integer `v` in big endian,
+    /// which is memory-comparable.
+    ///
+    /// This function is an alias to `write_u16`.
+    #[inline]
+    fn write_u16_be(&mut self, v: u16) -> Result<()> {
```

Comment:
Why adding the number logic?

Reply:
since write_u16_be is missing and it is used when generating sort key for general_ci.